metadata_layer.py

```diff
@@ -978,11 +979,13 @@ def apply(self, _db, collection, replace=None):
         if pool and self._availability_needs_update(pool):
             # Update availabily information. This may result in
             # the issuance of additional circulation events.
+            analytics = Analytics(_db)
```

Comment:
Yeah, this is a performance problem. We're looking at database queries and module imports every time we want to change a LicensePool. I think `analytics` should be passed into `ReplacementPolicy` just like `mirror` is.